jsm.go

```diff
@@ -384,10 +414,10 @@ func (js *js) DeleteConsumer(stream, consumer string, opts ...JSOpt) error {
 	}
 
 	if resp.Error != nil {
-		if resp.Error.Code == 404 {
+		if resp.Error.ErrorCode == ConsumerNotFound {
```

Comment:
Wonder if we could make these `ErrConsumerNotFound` plain error types, instead conform to the Error interface and maybe a JetStreamAPIError interface and stamp them with their error codes there, something like:

```
ErrConsumerNotFound             = &apiError{ErrorCode: ...}
```

Reply:
Good call. I have an idea on how to make the implementation a bit cleaner, but that would require changing the way errors are compared:
1. As you mentioned, change the defined error types to `APIError`, like so:
`ErrConsumerNotFound             = &APIError{ErrorCode: 10014, Code: 404, Description: "nats: consumer not found"}`
2. Implement custom `Is()` method on `APIError`, comparing error codes:
```
func (e *APIError) Is(target error) bool {
	err, ok := target.(*APIError)
	if !ok {
		return false
	}
	return err.ErrorCode == e.ErrorCode && err.Code == e.Code
}
```
3. Simplify client method implementations so that we do not have to compare the error codes manually at all, e.g.:
```
	if info.Error != nil {
		return nil, info.Error
	}
```
instead of
```
	if info.Error != nil {
		if info.Error.ErrorCode == ConsumerNotFound {
			return nil, ErrConsumerNotFound
		}
		if info.Error.ErrorCode == StreamNotFound {
			return nil, ErrStreamNotFound
		}
		return nil, info.Error
	}
```

However, that means user would not be able to check errors by simple comparison (`err == ErrConsumerNotFound`), but instead rely on `errors.Is(err, ErrConsumerNotFound)` (which usually should be the preferred way to go).

The other approach would be to leave the client method implementations untouched and simply change the variable definitions to be `&APIError{}` instead of `errors.New()`.

What do you think?

Reply:
Also, what do you mean by implementing `JetStreamAPIError` interface (i.e. what would it do). The `error` interface is implemented as part of this PR.

Reply:
I put together an example of what we could do via an interface here: https://github.com/nats-io/nats.go/pull/1047